modules/tensor_mechanics/src/materials/crystal_plasticity/ComputeMultipleCrystalPlasticityStress.C

```diff
@@ -446,10 +471,19 @@ ComputeMultipleCrystalPlasticityStress::calculateResidual()
   _inverse_plastic_deformation_grad =
       _inverse_plastic_deformation_grad_old * residual_equivalent_slip_increment;
 
-  _elastic_deformation_gradient =
-      _temporary_deformation_gradient * _inverse_plastic_deformation_grad;
+  // calculate F^{th -1}
+  Real dtheta = (_temperature[_qp] - _temperature_old[_qp]) * _substep_dt / _dt;
+  RankTwoTensor residual_equivalent_thermal_expansion_increment =
+      RankTwoTensor::Identity() - dtheta * _thermal_expansion_coefficients;
+  _inverse_thermal_deformation_grad =
+      _inverse_thermal_deformation_grad_old * residual_equivalent_thermal_expansion_increment;
+
+  _elastic_deformation_gradient = _temporary_deformation_gradient *
+                                  _inverse_thermal_deformation_grad *
+                                  _inverse_plastic_deformation_grad;
```

Comment:
Can we avoid those calculation if users do not use thermal eigenstrain?

Reply:
Fth is identity if temperature is a constant so I was assuming no/not much computation is added here. But yeah, I agree it is better to avoid the computation if no thermal coupling is specified